citesphere/src/main/java/edu/asu/diging/citesphere/web/user/EditItemController.java

```diff
@@ -43,8 +47,31 @@ public String showPage(Authentication authentication, Model model, CitationForm
         if (!model.containsAttribute("form")) {
             model.addAttribute("form", form);
         }
+        List<String> fields = new ArrayList<>();
+        citationManager.getItemTypeFields((IUser) authentication.getPrincipal(), citation.getItemType())
+            .forEach(f -> fields.add(f.getFilename()));
+        model.addAttribute("fields", fields);
         return "auth/group/items/item/edit";
     }
+
+    
+    /**
+     * Method to retrieve all fields filtered by item type.
+     */
+    @RequestMapping("/auth/group/{zoteroGroupId}/items/{itemId}/editItem")
+    public ResponseEntity<List<String>> updateItemType(Authentication authentication, Model model, CitationForm form, @PathVariable("zoteroGroupId") String zoteroGroupId, @PathVariable("itemId") String itemId,
+            @RequestParam(value = "itemType") ItemType itemType) {
+        ICitation citation = citationManager.getCitation((IUser)authentication.getPrincipal(), zoteroGroupId, itemId);
+        model.addAttribute("zoteroGroupId", zoteroGroupId);
+        model.addAttribute("citation", citation);
+        if (!model.containsAttribute("form")) {
+            model.addAttribute("form", form);
+        }
+        List<String> changedFields = new ArrayList<>();
```

Comment:
bad variable name; those fields have not been changed, they are just the fields that belong to the specified itemType.